src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -5103,6 +5115,46 @@ export class ResultsViewPageStore {
         },
     }));
 
+    public structuralVariantCache = new MobxPromiseCache<
+        { entrezGeneId: number },
+        StructuralVariant[]
+    >(q => ({
+        await: () => [
+            this.studyToStructuralVariantMolecularProfile,
+            this.studyToDataQueryFilter,
+        ],
+        invoke: async () => {
+            const studyIdToProfileMap = this
+                .studyToStructuralVariantMolecularProfile.result!;
+
+            if (_.isEmpty(studyIdToProfileMap)) {
+                return Promise.resolve([]);
+            }
+
+            const filters = this.samples.result.reduce(
+                (memo, sample: Sample) => {
```

Comment:
Do we need to specify this type `Sample` at here? Can't it be derived from the `this.samples.result`?

Reply:
Not required. It is mostly for us to know what type it is.

Reply:
Got it!